src/structures/paging/frame.rs

```diff
@@ -148,6 +148,18 @@ impl<S: PageSize> PhysFrameRange<S> {
     pub fn is_empty(&self) -> bool {
         self.start >= self.end
     }
+
+    /// Returns the number of frames in the range.
+    #[inline]
+    pub fn len(&self) -> u64 {
+        (self.end.start_address() - self.start.start_address()) / S::SIZE
+    }
```

Comment:
This will panic if `start` is greater than `end`.

```suggestion
    #[inline]
    pub fn len(&self) -> u64 {
        self.end - self.start
    }
```
(this doesn't fix the panic, but it makes the code simpler)